### PR TITLE
Update Composite Bow to correct stats and allow supported upgrades

### PR DESCRIPTION
**Pull request by euphemos**
**Summary**
- Initial Mana: per retail starts full (existing weenie starts empty)
- MeleeD Mod: 12% (existing weenie has BD7 applied)
- Allow handle upgrades (Handle -> Fine -> Superb -> Exquisite)
- Allow stat upgrades from pre-patch bows (6966) created prior to recipe updates in 2022.06 to post-patch (33997) bows by re-applying an intricately carved high undead (dark revenant thighbone) handle.
- Make upgraded bows a valid target for infinite ivory (note this is irrelevant due to Bug#1 below)

**Unfixed bugs in recipe (trying to keep this PR tight):**
1. Upgraded bows should not be pre-ivoried with the wieldable-by set, just ivoryable.
2. Only a perfectly crafted bow (6966) is eligible for new handles, but the recent recipe update allows crap bows to get the better stats. Source: memory and item description indicating a perfectly crafted (6966/33997) bow is required.

**Tests:**
```
@ci 6963 (perfectly crafted w/o handle)
@ci 6966 (perfectly crafted w/handle)
@ci 7055 (intricate/high dark rev)
@ci 33984 (fine bone handle)
@ci 33983 (superb bone handle)
@ci 33982 (exquisite bone handle)
@ci 30092 (infinite ivory)
```
Confirm pre-patch bow is upgradeable
Check spawned mana (500/500)
Check spawned MeleeD (12%)

Use 7055 (handle) on 6963 (no handle bow) - now a 33997 (bow w/handle)
Use 7055 (handle) on 6966 (pre-patch bow) - now a 33997 (bow w/handle)
  use 33984  on bow/handle - now bow/fine
  use 7055 on bow/fine (fails - no downgrade)
  use 33983  on bow/fine - now bow w/superb
  use 7055 on bow/superb (fails - no downgrade)
  use 33984 on bow/superb (fails - no downgrade)
  use 33982 on bow/superb - now bow w/exquisite
  use 7055 on bow/exquisite (fails - no downgrade)
  use 33984 on bow/exquisite (fails - no downgrade)
  use 33092 on bow/exquisite (fails - no downgrade)

**Supporting evidence:**

PCAPS:
Original: 6966 http://ac.yotesfan.com/weenies/items/6966
Post-Patch: 33997 http://ac.yotesfan.com/weenies/items/33997

2012 Video showing in-game stats https://youtu.be/3RsmCPwMNsI?t=648 also confirms item has full mana upon crafting as of Jan 2012.

Player notes:
https://acportalstorm.com/wiki/Composite_Bow#Composite_Bow_with_Handle
http://acpedia.org/wiki/Composite_Bow_with_Handle
https://asheron.fandom.com/wiki/Composite_Weapons_Quest
https://asheron.fandom.com/wiki/Composite_Bow_with_Handle#Composite_Bow_with_Handle_(High)
https://asheron.fandom.com/wiki/Bone_Handle_(High)
http://acpedia.org/wiki/Missile_Weaponry/Bows
http://acpedia.org/wiki/Composite_Bow (incorrectly says bow 250 vs 240)

**Perfectly crafted component improvements:**

Shaped Great Mattekar Horn	+6% Melee Defense
Cured Large Lugian Sinew	Bow: Speed 35
Excellent Oiled String	Bow: +110% Damage

Bone Handle (High) (successful Dark Rev)
Made from intricately carving a Dark Revenant Thighbone.
Can be attached to a Composite Bow, Composite Crossbow, or Composite Atlatl
Improves weapon by adding +6% to melee, +25% to damage modifier (bow), Armor Cleaving, Cast on Strike, Brittlemail VI, Defender VI, Boon of Refinement, Swift Killer VI, Strathelar's Boon, Infected Caress. It also makes the weapon heavier (1,520 burden instead of 980 for bow)
Requirements: Arcane Lore skill 170+, Bow Skill 240+.

**Correct un-buffed final Composite Bow with Handle:**

Value: 400
Burden: 1,520
Skill: Missile Weapons (Bow)
Damage Bonus: 0
Damage Modifier: +135%
Speed: Average (35)
Range: 80 yds
Uses arrows as ammunition.
Bonus to Melee Defense: +12%
Spells: Brittlemail VI, Aura of Defender VI, Boon of Refinement, Swift Killer VI, Missile Weapon Mastery Other VII, Aura of Infected Caress
Properties: Armor Cleaving, Cast on Strike, Attuned, Bonded, Ivoryable
Wield requires Missile Weapons 240
Activation requires Arcane Lore: 170
This item cannot be sold.
Spellcraft: 250
Mana: 500
Mana Cost: 1 point per 20 seconds

### CODE REVIEW

**Comment by ziang4891** [comment]
No to this PR.  You are attempting to add things that are not supported by the data; also, Blood Drinker does not affect missile weapons for buffing damage

**Comment by ziang4891** [comment]
Wiki is not proof of anything

File: Database/Patches/4 CraftTable/01195 Composite Bow with Handle.sql [hunk 1]
```diff
@@ -17,4 +17,5 @@ VALUES (@parent_id, 3,  25, NULL, 4, 0) /* On Player.SuccessResult CopyFromSourc
 DELETE FROM `cook_book` WHERE `recipe_Id` = 1195;
 
 INSERT INTO `cook_book` (`recipe_Id`, `source_W_C_I_D`, `target_W_C_I_D`, `last_Modified`)
-VALUES (1195, 7055 /* Bone Handle */,  6963 /* Composite Bow */, '2022-06-21 15:22:25');
+VALUES (1195, 7055 /* Bone Handle */,  6963 /* Composite Bow */, '2022-06-21 15:22:25')
+     , (1195, 7055 /* Bone Handle */,  6966 /* Composite Bow with Handle (pre-patch) */, '2022-06-24 13:42:42');
```
Comment (ziang4891) [inline Database/Patches/4 CraftTable/01195 Composite Bow with Handle.sql:21]: Shouldn't be able to upgrade a pre-patch, completed bow

File: Database/Patches/4 CraftTable/05300 Composite Bow with Fine Handle.sql [hunk 2]
```diff
@@ -35,4 +35,5 @@ VALUES (5300, 33984 /* Fine Bone Handle */,  6895 /* Composite Bow */, '2022-06-
      , (5300, 33984 /* Fine Bone Handle */,  6955 /* Composite Bow */, '2022-06-21 15:22:25')
      , (5300, 33984 /* Fine Bone Handle */,  6959 /* Composite Bow */, '2022-06-21 15:22:25')
      , (5300, 33984 /* Fine Bone Handle */,  6963 /* Composite Bow */, '2022-06-21 15:22:25')
-     , (5300, 33984 /* Fine Bone Handle */, 33997 /* Composite Bow with Handle */, '2022-06-21 15:22:25');
+     , (5300, 33984 /* Fine Bone Handle */,  6966 /* Composite Bow with Handle (pre-patch) */, '2022-06-23 16:20:48')
+     , (5300, 33984 /* Fine Bone Handle */, 33997 /* Composite Bow with Handle */, '2022-06-23 16:20:48');
```
Comment (ziang4891) [inline Database/Patches/4 CraftTable/05300 Composite Bow with Fine Handle.sql:39]: It should only be possible to upgrade the 33997.  All of the 6k wcids need to be removed.  The 33997 should be the only entry in the cookbook

File: Database/Patches/9 WeenieDefaults/MissileLauncher/MissileWeapon/33997 Composite Bow with Handle.sql [hunk 3]
```diff
@@ -23,15 +23,15 @@ VALUES (33997,   1,        256) /* ItemType - MissileWeapon */
      , (33997,  60,        192) /* WeaponRange */
      , (33997,  93,       1044) /* PhysicsState - Ethereal, IgnoreCollisions, Gravity */
      , (33997, 106,        250) /* ItemSpellcraft */
-     , (33997, 107,          0) /* ItemCurMana */
+     , (33997, 107,        500) /* ItemCurMana */
```
Comment (ziang4891) [inline Database/Patches/9 WeenieDefaults/MissileLauncher/MissileWeapon/33997 Composite Bow with Handle.sql:26]: This item should not have any starting mana

File: Database/Patches/9 WeenieDefaults/MissileLauncher/MissileWeapon/33997 Composite Bow with Handle.sql [hunk 4]
```diff
@@ -23,15 +23,15 @@ VALUES (33997,   1,        256) /* ItemType - MissileWeapon */
...
      , (33997, 108,        500) /* ItemMaxMana */
      , (33997, 109,        170) /* ItemDifficulty */
      , (33997, 114,          1) /* Attuned - Attuned */
      , (33997, 150,        103) /* HookPlacement - Hook */
      , (33997, 151,          2) /* HookType - Wall */
      , (33997, 158,          2) /* WieldRequirements - RawSkill */
      , (33997, 159,         47) /* WieldSkillType - MissileWeapons */
-     , (33997, 160,        250) /* WieldDifficulty */
+     , (33997, 160,        240) /* WieldDifficulty */
```
Comment (ziang4891) [inline Database/Patches/9 WeenieDefaults/MissileLauncher/MissileWeapon/33997 Composite Bow with Handle.sql:34]: This change is incorrect, per the PCAP'ed data for this item.  It is supposed to be a 250 wield difficulty

**Review by ziang4891 — CHANGES_REQUESTED** [review]

**Comment by euphemos** [comment]
Withdrawing this, and will submit bug reports. When we reach agreement on the changes I'll be happy to submit those.

**Pull request closed without merging**